internal/dag/builder.go

```diff
@@ -308,17 +308,18 @@ func isBlank(s string) bool {
 	return len(strings.TrimSpace(s)) == 0
 }
 
-// minProtoVersion returns the TLS protocol version specified by an ingress annotation
+// MinProtoVersion returns the TLS protocol version specified by an ingress annotation
```

Comment:
if we can set the min Version via flag/config during startup we can contain the logic around this function.. i.e return a version equal or higher then the one configured during startup..

Reply:
const (
        DEFAULT_MIN_TLS_VERSION = auth.TlsParameters_TLSv1_1
)

var minTlsProtocol auth.TlsParameters_TlsProtocol = DEFAULT_MIN_TLS_VERSION

var tlsProtocols = map[string]auth.TlsParameters_TlsProtocol {
        "1.1": auth.TlsParameters_TLSv1_1,
        "1.2": auth.TlsParameters_TLSv1_2,
        "1.3": auth.TlsParameters_TLSv1_3,
}

func SetClusterMinTlsVersion(proto string) {
        p, ok := tlsProtocols[proto]
        if !ok {
                p = DEFAULT_MIN_TLS_VERSION
        }
        minTlsProtocol = p
}
func getClusterMinTlsVersion() auth.TlsParameters_TlsProtocol {
        return minTlsProtocol
}

func minProtoVersion(version string) auth.TlsParameters_TlsProtocol {
        p, ok := tlsProtocols[version]
        if !ok || p < getClusterMinTlsVersion() {
                p = getClusterMinTlsVersion()
        }
        return p
}